src-tauri/src/gui/commands/installation.rs

```diff
@@ -253,8 +253,23 @@ app_handle: &AppHandle,
         }
     });
 
-    let default_mirror = rust_i18n::t!("gui.installation.default_mirror").to_string();
-    let mirror = settings.idf_mirror.as_deref().unwrap_or(&default_mirror);
+    let default_mirror_str = rust_i18n::t!("gui.installation.default_mirror").to_string();
+    let default_mirror = settings
+        .idf_mirror
+        .as_deref()
+        .unwrap_or(&default_mirror_str);
+    let mirror_latency_map = settings.get_idf_mirror_latency_map().await.unwrap();
+    let best_mirror = mirror_latency_map
+        .iter()
+        .min_by_key(|(_, latency)| *latency)
+        .unwrap()
```

Comment:
this is duplicated logic from the idf_tools. ... it's duplicated error which will cause the app to crash.